src/main/java/com/openclassrooms/safetynet/mapper/PersonInfoDTOMapper.java

```diff
@@ -0,0 +1,36 @@
+package com.openclassrooms.safetynet.mapper;
+
+import org.springframework.stereotype.Component;
+
+import com.openclassrooms.safetynet.DTO.PersonInfoDTO;
+import com.openclassrooms.safetynet.model.Allergy;
+import com.openclassrooms.safetynet.model.Medication;
+import com.openclassrooms.safetynet.model.Person;
+
+@Component
+public class PersonInfoDTOMapper {
+
+	public PersonInfoDTO convertPersonToPersonInfoDTO(Person person) {
+
+		String medications = "";
+		String allergies = "";
+		for (Medication medication : person.getMedicalRecord().getMedication()) {
+			if (medications == "") {
+				medications = medication.getMedication();
+			} else {
+				medications = medications + ", " + medication.getMedication();
+			}
+		}
+		for (Allergy allergy : person.getMedicalRecord().getAllergy()) {
+			if (allergies == "") {
+				allergies = allergy.getAllergyName();
+			} else {
+				allergies = allergies + ", " + allergy.getAllergyName();
+			}
+		}
```

Comment:
```suggestion
                 String medications = String.join(", ", person.getMedicalRecord().getMedication());
		String allergies = String.join(", ", person.getMedicalRecord().getAllergy());
```

Reply:
probleme avec la methode join() car person.getMedicalRecord().getAllergy() par exemple, est une liste d'objet Allergy.
Il faut necessairement une boucle for each